Bottom: 4ace8a5af825d66049eb065fa2a01e9714be081a
Top:    8b5fff8383febcc289012f202526cc0b950bf59e
Author: Ian Kent <raven@themaw.net>
Date:   2013-08-02 19:13:54 +0800

Refresh of autofs-5.0.7-fix-fix-wildcard-multi-map-regression.patch

---

diff --git a/modules/lookup_sss.c b/modules/lookup_sss.c
index 6bf9eb4..297b3b2 100644
--- a/modules/lookup_sss.c
+++ b/modules/lookup_sss.c
@@ -639,9 +639,9 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 	 * We can't take the writelock for direct mounts. If we're
 	 * starting up or trying to re-connect to an existing direct
 	 * mount we could be iterating through the map entries with
+	 * mount we could be iterating through the map entries with
 	 * the readlock held. But we don't need to update the cache
 	 * when we're starting up so just take the readlock in that
-	 * case.
 	 */
 	if (ap->flags & MOUNT_FLAG_REMOUNT)
 		cache_writelock(mc);
@@ -659,6 +659,7 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 	if (me && me->mapent) {
 		/*
 		 * If this is a lookup add wildcard match for later validation
+		 * checks and negative cache lookups.
 		 */
 		if (ap->type == LKP_INDIRECT && *me->key == '*' &&
 		   !(ap->flags & MOUNT_FLAG_REMOUNT)) {
